LoopKitUI/OnboardingUI.swift

```diff
@@ -0,0 +1,87 @@
+    ///
+    /// - Parameters:
+    ///     - identifier: The identifier of the pump manager to setup.
+    /// - Returns: Either a conforming view controller to create and setup the pump manager, a newly created and setup pump manager, or an error.
+    func setupPumpManager(withIdentifier identifier: String, initialSettings settings: PumpManagerSettings) -> UIResult<UIViewController & PumpManagerCreateNotifying & PumpManagerOnboardNotifying & CompletionNotifying, PumpManager, Error>
+}
+
+public protocol ServiceProvider: AnyObject {
+    /// The active services.
+    var activeServices: [Service] { get }
+
+    /// This descriptor list of available services.
+    var availableServices: [ServiceDescriptor] { get }
+
+    /// Setup the setup with the specified identifier.
+    ///
+    /// - Parameters:
+    ///     - identifier: The identifier of the setup to setup.
+    /// - Returns: Either a conforming view controller to create and setup the service, a newly created and setup service, or an error.
+    func setupService(withIdentifier identifier: String) -> UIResult<UIViewController & ServiceCreateNotifying & ServiceOnboardNotifying & CompletionNotifying, Service, Error>
+}
+
+public protocol OnboardingDelegate: AnyObject {
+    /// Informs the delegate that onboarding has new therapy settings that should be persisted.
+    ///
+    /// - Parameters:
+    ///     - notifying: The source of the notification.
+    ///     - therapySettings: The new therapy settings.
+    func onboardingNotifying(_ notifying: OnboardingNotifying, hasNewTherapySettings therapySettings: TherapySettings)
+}
+
+public protocol OnboardingNotifying: AnyObject {
+    /// Delegate to notify about onboarding changes.
+    var onboardingDelegate: OnboardingDelegate? { get set }
+}
+
+public protocol OnboardingUI: Onboarding {
+    /// Provides a view controller to configure onboarding, if needed.
+    ///
+    /// - Parameters:
+    ///   - preferredGlucoseUnit: The preferred glucose unit.
+    ///   - cgmManagerProvider: The provider of CGM Manager functionality.
+    ///   - pumpManagerProvider: The provider of Pump Manager functionality.
+    ///   - serviceProvider: The provider of Service functionality.
+    ///   - colorPalette: The colors to use in any UI,
+    /// - Returns: A view controller to create and configure a new onboarding.
+    func onboardingViewController(preferredGlucoseUnit: HKUnit,
+                                  cgmManagerProvider: CGMManagerProvider,
+                                  pumpManagerProvider: PumpManagerProvider,
+                                  serviceProvider: ServiceProvider,
+                                  colorPalette: LoopUIColorPalette) -> (UIViewController & OnboardingNotifying & CGMManagerCreateNotifying & CGMManagerOnboardNotifying & PumpManagerCreateNotifying & PumpManagerOnboardNotifying & ServiceCreateNotifying & ServiceOnboardNotifying & CompletionNotifying)
```

Comment:
[nit] consider `typealias`ing that long list of protocol conformance requirements to something more palatable?  May I suggest:

```
typealias OnboardingViewController = UIViewController & OnboardingNotifying & CGMManagerCreateNotifying & CGMManagerOnboardNotifying & PumpManagerCreateNotifying & PumpManagerOnboardNotifying & ServiceCreateNotifying & ServiceOnboardNotifying & CompletionNotifying
```

Reply:
I thought about that, but it wasn't consistent with other code that has long lists of protocol requirements. I can add it here if you think it makes a big difference.

Reply:
I think it isn't that bad to be inconsistent, since there are so many.  Also, consider not putting them all on one line:

```swift
typealias OnboardingViewController = UIViewController & 
    OnboardingNotifying & 
    CGMManagerCreateNotifying & 
    CGMManagerOnboardNotifying & 
    PumpManagerCreateNotifying & 
    PumpManagerOnboardNotifying & 
    ServiceCreateNotifying & 
    ServiceOnboardNotifying & 
    CompletionNotifying
```

Reply:
...or, another idea:

```swift
typealias AllNotifying = OnboardingNotifying & 
    CGMManagerCreateNotifying & 
    CGMManagerOnboardNotifying & 
    PumpManagerCreateNotifying & 
    PumpManagerOnboardNotifying & 
    ServiceCreateNotifying & 
    ServiceOnboardNotifying & 
    CompletionNotifying
```
And then just have this function return `(UIViewController & AllNotifying)`

Reply:
I think I'll go with the `OnboardingViewController` alias above. :+1: